src/internal/hooks/use-virtual/index.ts

```diff
@@ -34,8 +36,15 @@ export function useVirtual<Item extends object>({
   items,
   parentRef,
   estimateSize,
+  firstItemSticky,
 }: UseVirtualProps<Item>): RowVirtualizer {
-  const rowVirtualizer = useVirtualDefault({ size: items.length, parentRef, estimateSize, overscan: 5 });
+  const rowVirtualizer = useVirtualDefault({
+    size: items.length,
+    parentRef,
+    estimateSize,
+    overscan: 5,
+    rangeExtractor: firstItemSticky ? stickyRangeExtractor : undefined,
```

Comment:
React-Virtual docs for `rangeExtractor`: https://tanstack.com/virtual/latest/docs/api/virtualizer#rangeextractor

Reply:
This is necessary to make sure that the sticky item (with index 0) is always present in the DOM, despite scrolling down with the virtual list.